ResearchKit/Common/ORKReviewStepViewController.m

```diff
@@ -0,0 +1,213 @@
+}
+
+- (void)setSkipButtonItem:(UIBarButtonItem *)skipButtonItem {
+    [super setSkipButtonItem:skipButtonItem];
+    _continueSkipView.skipButtonItem = self.skipButtonItem;
+}
+
+- (void)stepDidChange {
+    [super stepDidChange];
+    
+    [_tableContainer removeFromSuperview];
+    _tableContainer = nil;
+    
+    _tableContainer.tableView.delegate = nil;
+    _tableContainer.tableView.dataSource = nil;
+    _continueSkipView = nil;
+    
+    if ([self reviewStep]) {
+        _tableContainer = [[ORKTableContainerView alloc] initWithFrame:self.view.bounds];
+        _tableContainer.tableView.delegate = self;
+        _tableContainer.tableView.dataSource = self;
+        _tableContainer.tableView.clipsToBounds = YES;
+
+        [self.view addSubview:_tableContainer];
+        _tableContainer.tapOffView = self.view;
+        
+        _tableContainer.stepHeaderView.captionLabel.useSurveyMode = self.step.useSurveyMode;
+        _tableContainer.stepHeaderView.captionLabel.text = [self reviewStep].title;
+        _tableContainer.stepHeaderView.instructionLabel.text = [self reviewStep].text;
+        _tableContainer.stepHeaderView.learnMoreButtonItem = self.learnMoreButtonItem;
+        
+        _continueSkipView = _tableContainer.continueSkipContainerView;
+        _continueSkipView.skipButtonItem = self.skipButtonItem;
+        _continueSkipView.continueEnabled = YES;
+        _continueSkipView.continueButtonItem = self.continueButtonItem;
+        _continueSkipView.optional = self.step.optional;
+        [_tableContainer setNeedsLayout];
+    }
+}
+
+- (ORKReviewStep *)reviewStep {
+    return [self.step isKindOfClass:[ORKReviewStep class]] ? (ORKReviewStep *) self.step : nil;
+}
+
+//TODO: state restoration
+
+#pragma mark UITableViewDataSource
+
+- (NSInteger)numberOfSectionsInTableView:(nonnull UITableView *)tableView {
+    return _steps.count > 0 ? ORKReviewSectionCount : 0;
+}
+
+- (NSInteger)tableView:(nonnull UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return (section == ORKReviewSectionSpace1 || section == ORKReviewSectionSpace2) ? 1 : _steps.count;
+}
+
+- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
+    tableView.layoutMargins = UIEdgeInsetsZero;
+    if (indexPath.section == ORKReviewSectionSpace1 || indexPath.section == ORKReviewSectionSpace2) {
+        static NSString *SpaceIdentifier = @"Space";
+        UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:SpaceIdentifier];
+        if (cell == nil) {
+            cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:SpaceIdentifier];
+        }
+        return cell;
+    }
+    static NSString *identifier = nil;
+    identifier = [NSStringFromClass([self class]) stringByAppendingFormat:@"%@", @(indexPath.row)];
+    ORKChoiceViewCell *cell = [tableView dequeueReusableCellWithIdentifier:identifier];
```

Comment:
I'd like to keep ORKChoiceCell since we want to display a human readable version of the step result which might span across multiple lines.